test/merge-patient-lab-iips-mocha.js

```diff
@@ -5,151 +5,260 @@ var chaiAsPromised = require('chai-as-promised');
+
+            describe('#errorState', function() {
+
+                it("should initially return nothing", function() {
+                    return Promise.resolve(componentFactory.getComponent )
+                        .then(commonTest.createComponent).then(function(component){
+                            return component.rpf.vni('').errorState();
+                    }).should.become(undefined);
+                });
             });
-            it("should have patient input state data after input", function() {
-                return Promise.resolve(createComponent()).then(function(component){
-                    var labwork = noflo.internalSocket.createSocket();
-                    component.inPorts.labwork.attach(labwork);
-                    labwork.send({id: '001',  glucose: '75',  date: '2012-02-01'});
-                    labwork.disconnect();
-                    component.inPorts.patient.detach(labwork);
-                    return component;
-                }).then(promiseLater).then(function(component){
-                    return component.rpf.createVni('').getInputState('labwork', '', '');
-                }).then(_.property('data')).should.become({id: '001',  glucose: '75',  date: '2012-02-01'});
+
+            describe('#inputState', function() {
+
+                it("should initially return nothing", function() {
+                    return Promise.resolve(componentFactory.getComponent )
+                        .then(commonTest.createComponent).then(function(component){
+                            return component.rpf.vni('').inputState('patient');
+                    }).should.become(undefined);
+                });
+
+                it("should have input state after input", function() {
+                    return Promise.resolve(componentFactory.getComponent )
+                        .then(commonTest.createComponent).then(function(component){
+
+                            // initialize state
+                            component.rpf.vni('').inputState('patient',
+                                                             { id: '001', 
+                                                               name: 'Alice', 
+                                                               dob: '1979-01-23' });
+
+                            var state = component.rpf.vni('').inputState('patient');
+                            state.should.be.an('object');
+                            state.should.have.all.keys( 'data', 'lm', 'previousLms', 'port' );
+                    });
+                });
+
+                it("should have input state data after input", function() {
+                    return Promise.resolve(componentFactory.getComponent )
+                        .then(commonTest.createComponent).then(function(component){
+
+                            // initialize state
+                            var patientState = { id: '001', name: 'Alice', dob: '1979-01-23' };
```

Comment:
The reason is that at the underlying Framework level, it must be possible to get/set an entire state object -- including data and lm properties -- not just the data part of it.  That is because an updater may need to be able to update the data part *without* updating the lm.  It might do this, for example, if it decided to change the internal representation of the data but did not want to cause an event to be triggered notifying its downstream neighbors.